src/App.tsx

```diff
@@ -730,126 +775,94 @@ function App(): ReactElement {
-            >
-              <CanvasWrapper
-                canv={canv}
-                dataset={dataset}
-                selectedBackdropKey={selectedBackdropKey}
-                colorRamp={getColorMap(colorRampData, colorRampKey, colorRampReversed)}
-                colorRampMin={colorRampMin}
-                colorRampMax={colorRampMax}
-                categoricalColors={categoricalPalette}
-                selectedTrack={selectedTrack}
-                config={config}
-                onTrackClicked={(track) => {
-                  setFindTrackInput("");
-                  setSelectedTrack(track);
-                }}
-                inRangeLUT={inRangeLUT}
-                onMouseHover={(id: number): void => {
-                  const isObject = id !== BACKGROUND_ID;
-                  setShowHoveredId(isObject);
-                  if (isObject) {
-                    setLastHoveredId(id);
-                  }
-                }}
-                onMouseLeave={() => setShowHoveredId(false)}
-              />
-            </HoverTooltip>
+            <div className={styles.canvasTopAndCanvasContainer}>
+              <div className={styles.canvasTopContainer}>
+                <h3 style={{ margin: "0" }}>
+                  {dataset ? dataset.getFeatureNameWithUnits(featureName) : "Feature value range"}
+                </h3>
+                <FlexRowAlignCenter $gap={12} style={{ flexWrap: "wrap", justifyContent: "space-between" }}>
+                  <div style={{ flexBasis: 250, flexShrink: 2, flexGrow: 2, minWidth: "75px" }}>
+                    {
+                      // Render either a categorical color picker or a range slider depending on the feature type
+                      dataset?.isFeatureCategorical(featureName) ? (
+                        <CategoricalColorPicker
+                          categories={dataset.getFeatureCategories(featureName) || []}
+                          selectedPalette={categoricalPalette}
+                          onChangePalette={setCategoricalPalette}
+                          disabled={disableUi}
+                        />
+                      ) : (
+                        <LabeledRangeSlider
+                          min={colorRampMin}
+                          max={colorRampMax}
+                          minSliderBound={dataset?.getFeatureData(featureName)?.min}
+                          maxSliderBound={dataset?.getFeatureData(featureName)?.max}
+                          onChange={function (min: number, max: number): void {
+                            setColorRampMin(min);
+                            setColorRampMax(max);
+                          }}
+                          marks={getColorMapSliderMarks()}
+                          disabled={disableUi}
+                        />
+                      )
+                    }
+                  </div>
+                  <div style={{ flexBasis: 100, flexShrink: 1, flexGrow: 1, width: "fit-content" }}>
```

Comment:
you seem to have some custom components to handle flex settings. why is this a div with inline styling?

Reply:
I had these inline because I was making a lot of tweaks to them to try and get the flexbox to behave correctly. I want to refactor `App` in the future to use CSS-in-JS using styled components, so my preference has been towards either inline styling or styled-components rather than adding them to the CSS module file.